src/tilda.c

```diff
@@ -269,18 +290,29 @@ int main (int argc, char *argv[])
     } else {    // if not, we look for the defaut config file
         config_file = get_config_file_name (lock.instance);
     }
+    /* Start up the configuration system and load from file */
+    gint config_init_result = config_init (config_file);
+
+    /* Set up possible overridden config options */
+    setup_config_from_cli_options (cli_options);
+
+    if (cli_options->toggle_win_visibility) {
+        char window_id = (cli_options->toggle_win_visibility) + '0' - 1; // type default is 0
```

Comment:
This is hard to understand, we are adding a boolean with a character (`'0'`) and then subtract one. The ASCII value of `'0'` is 48, so this gives either 47 or 48, depending if `cli_options->toggle_win_visibility` has a value of 0 or 1.

I guess we may need a value here, that is passed from the command line to point to the DBUS bus instance of the target process to which we want to send the command to. This would usually be a value from 1 to N, but not something like 47. 

Am I missing something here? Does this actually work?

Reply:
now, i wish i remembered why i wrote it this way too :) i remember it did work, but anyways, indeed playing with implicit type conversions in C was a nice method of shooting yourself in foot. I'll take a look at this while testing with never ubu, i'm curious if this still works myself.

Reply:
Yeah, C has a lot of funny ways to convert between different data types, but a lot of that is error prone. I would try to avoid this and just use one of the `printf` family functions, like `snprintf` to combine a string template with an integer value.